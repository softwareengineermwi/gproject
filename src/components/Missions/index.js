import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Mission from './Mission';
import { getListOfMissions } from '../../redux/missions/missions';
import Styles from './mission.module.scss';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (missions.length === 0) dispatch(getListOfMissions());
  }, [dispatch, missions.length]);

  return (
    <div className="container margin-y-sm">
      <table className={Styles.missionsTable}>
        <thead>
          <tr>
            <td>Mission</td>
            <td>Description</td>
            <td>Status</td>
            <td>Join</td>
          </tr>
        </thead>
        <tbody className="tbody">
          {
            missions.map((mission) => (<Mission key={mission.id} missionInfo={mission} />))
          }
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
