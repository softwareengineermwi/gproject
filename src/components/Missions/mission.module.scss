@mixin badge ($color: DarkGray) {
  display: block;
  width: 8rem;
  padding: 2px;
  text-align: center;
  background-color: $color;
  color: white;
  font-weight: bold;
  border-radius: 10px;
  -webkit-border-radius: 10px;
  -moz-border-radius: 10px;
  -ms-border-radius: 10px;
  -o-border-radius: 10px;
}

.activeMember {
  @include badge($color: #18a2b8);
}

.notMember {
  @include badge();
}

@mixin joinBtn ($color: rgb(100, 100, 100)) {
  width: 8rem;
  padding: 5px;
  font-size: 1rem;
  color: $color;
  border: 2px solid $color;
  border-radius: 6px;
  -webkit-border-radius: 6px;
  -moz-border-radius: 6px;
  -ms-border-radius: 6px;
  -o-border-radius: 6px;
}

.joinMission {
  @include joinBtn ();
}

.leaveMission {
  @include joinBtn ($color: rgb(231, 43, 106));
}
