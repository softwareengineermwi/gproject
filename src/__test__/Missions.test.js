import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/Missions';

describe('Missions Component test', () => {
  it('Should render correctly', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
