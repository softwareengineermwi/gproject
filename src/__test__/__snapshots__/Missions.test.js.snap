// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Component test Should render correctly 1`] = `
<div
  className="container margin-y-sm"
>
  <table
    className="missionsTable"
  >
    <thead>
      <tr>
        <td>
          Mission
        </td>
        <td>
          Description
        </td>
        <td>
          Status
        </td>
        <td>
          Join
        </td>
      </tr>
    </thead>
    <tbody
      className="tbody"
    />
  </table>
</div>
`;
